src/System.Linq.Parallel/src/System/Linq/Parallel/Merging/OrderPreservingMergeHelper.cs

```diff
@@ -45,7 +45,7 @@ internal class OrderPreservingMergeHelper<TInputOutput, TKey> : IMergeHelper<TIn
 
             _taskGroupState = new QueryTaskGroupState(cancellationState, queryId);
             _partitions = partitions;
-            _results = new Shared<TInputOutput[]>(null);
+            _results = new Shared<TInputOutput[]>(null!);
```

Comment:
Shouldn't `_results` be defined instead as `Shared<TInputOutput[]?>`?

Reply:
that's better thanks!